src/Sulu/Bundle/AdminBundle/Resources/js/containers/Badge/stores/BadgeStore.js

```diff
@@ -0,0 +1,122 @@
+// @flow
+import {action, reaction, observable, computed} from 'mobx';
+import jsonPointer from 'json-pointer';
+import debounce from 'debounce';
+import symfonyRouting from 'fos-jsrouting/router';
+import Router from '../../../services/Router';
+import Requester from '../../../services/Requester';
+import type {HandleResponseHook} from '../../../services/Requester/types';
+
+export default class BadgeStore {
+    router: Router;
+    routeName: string;
+    dataPath: ?string;
+    requestParameters: Object;
+    routerAttributesToRequest: Object;
+    @observable value: ?string = null;
+    disposer: () => {};
+
+    constructor(
+        router: Router,
+        routeName: string,
+        dataPath: ?string,
+        requestParameters: Object,
+        routerAttributesToRequest: Object
+    ) {
+        this.router = router;
+        this.routeName = routeName;
+        this.dataPath = dataPath;
+        this.requestParameters = requestParameters;
+        this.routerAttributesToRequest = routerAttributesToRequest;
+    }
+
+    @computed get evaluatedRequestParameters() {
+        const {
+            router: {
+                attributes: routerAttributes,
+            },
+            requestParameters: attributesToRequest,
+            routerAttributesToRequest,
+        } = this;
+
+        const requestParameters = {};
+        Object.keys(routerAttributesToRequest)
+            .forEach((routerAttributeKey) => {
+                const requestAttributeKey = routerAttributesToRequest[routerAttributeKey];
+                const attributeName = isNaN(routerAttributeKey)
+                    ? routerAttributeKey
+                    : requestAttributeKey;
+
+                requestParameters[requestAttributeKey] = routerAttributes[attributeName];
+            });
+
+        return {...requestParameters, ...attributesToRequest};
+    }
+
+    @computed get url() {
+        const {routeName} = this;
+
+        return symfonyRouting.generate(routeName, this.evaluatedRequestParameters);
+    }
+
+    @action setData(data: any) {
+        const {dataPath} = this;
+
+        let enhancedData = data;
+        if (dataPath) {
+            enhancedData = jsonPointer.get(data, dataPath);
+        }
+
+        this.value = String(enhancedData);
+    }
+
+    load = debounce(() => {
+        Requester.get(this.url).then((response: Object) => {
+            this.setData(response);
+        });
+    }, 3000, true);
```

Comment:
isnt this a bit long?

Reply:
The `true` at the end executes the request immediately and stops all following requests in the next 3 seconds, so I think this should be fine

Reply:
okay 🙂